components/contactUs.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useRef, useEffect } from "react";
+import { RenderMsg } from "./renderMsg";
+
+type FormState = {
+  name: string;
+  email: string;
+  message: string;
+};
+
+export const ContactFrom = () => {
+  const [nameErr, setNameErr] = useState("");
+  const [emailErr, setEmailErr] = useState("");
+  const [messageErr, setMessageErr] = useState("");
+  const [state, setstate] = useState<FormState>({
+    name: "",
+    email: "",
+    message: "",
+  });
+
+  const nameInput = useRef<HTMLInputElement>(null);
+  const emailInput = useRef<HTMLInputElement>(null);
+  const messageInput = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    if (nameInput.current != null) {
+      nameInput.current.focus();
+    }
+  }, []);
+
+  const setFocus = (
+    ref:
+      | React.RefObject<HTMLInputElement>
+      | React.RefObject<HTMLTextAreaElement>
+  ) => {
+    if (ref.current != null) {
+      ref.current.focus();
+    }
+  };
+
+  const handleChange = (
+    event:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setstate({ ...state, [event.target.name]: event.target.value });
+  };
+
+  const formIsValid = () => {
+    let validRegex =
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+
+    if (state.name.length === 0) {
```

Comment:
Although here it isn't as dire, because you only have 3 fields, but the more fields you have in the future - the more overloaded this function will become.

I'd suggest using a more declarative approach: separate out your validation logic into separate small functions (like `isNameValid(val: string): boolean` etc.) into a different file (like validations.ts). Then, you can import those functions and use them here. The important thing is that here you don't care about the implementation, which is a more declarative approach.

Reply:
Actually, your validation functions can even be better: don't return just true/false, instead return a `string` containing the error message, if there is an error, and `null` if there are no errors.

That way - here you can just call `setNameErr(validateName(state.name))` and it will just work.